src/code/padmgr.c

```diff
@@ -298,15 +398,19 @@ void PadMgr_HandleRetraceMsg(PadMgr* padMgr) {
     padMgr->validCtrlrsMask = mask;
 
     if (gFaultMgr.msgId != 0) {
+        // If fault is active, no rumble
         PadMgr_RumbleStop(padMgr);
-    } else if (padMgr->rumbleOffFrames > 0) {
-        --padMgr->rumbleOffFrames;
+    } else if (padMgr->rumbleOffTimer > 0) {
+        // If the rumble off timer is active, no rumble
+        --padMgr->rumbleOffTimer;
         PadMgr_RumbleStop(padMgr);
-    } else if (padMgr->rumbleOnFrames == 0) {
+    } else if (padMgr->rumbleOnTimer == 0) {
+        // If the rumble on timer is inactive, no rumble
         PadMgr_RumbleStop(padMgr);
     } else if (!padMgr->preNMIShutdown) {
-        PadMgr_RumbleControl(padMgr);
-        --padMgr->rumbleOnFrames;
+        // If not in PreNMI, update rumble
```

Comment:
```suggestion
        // If not past PreNMI, update rumble
```
given that it's a single point in time, right (I get what "in prenmi" would mean but exactitude y'know)